app/logic/trackAttendees.py

```diff
@@ -0,0 +1,13 @@
+from app.models.eventParticipant import EventParticipant
+from app.models.user import User
+from app.models.event import Event
+from peewee import fn
+
+def trainedParticipants(programID, trlist):
+    """
+    This function tracks the users who have attended every Prerequisite
+    event and adds them to a list that will not flag them when tracking hours.
+    """
+    eventTrainingDataList = [participant.user.username for participant in (EventParticipant.select().where(EventParticipant.event.in_(trlist)))]
+    attendedTraining = list(dict.fromkeys(filter(lambda user: eventTrainingDataList.count(user) == len(trlist), eventTrainingDataList)))

```

Comment:
This is fancy, but really seems like it could be simplified into one query with the calling code. `Select all users who do not have a missing event participant record for any training event in the program`

Reply:
we aren't going to worry about this now